spec.md

```diff
@@ -499,6 +499,56 @@ the module.  The `c`{.firrtl} sub-field contained in the `b`{.firrtl} sub-field
 flows into the module, and the `d`{.firrtl} sub-field contained in the
 `b`{.firrtl} sub-field flows out of the module.
 
+## Type Modifiers
+
+### Constant Type
+
+A constant type is a type whose value is guaranteed to be unchanging at circuit 
+execution time.  Constant is a constraint on the mutability of the value, it 
+does not imply a literal value at a point in the emitted design.  Constant types 
+may be used in ports, wire, nodes, and generally anywhere a non-constant type is 
+usable.  Operations on constant type are well defined.  As a general rule (with 
+any exception listed in the definition for such operations as have exceptions), 
+an operation whose arguments are constant produces a constant.  An operations 
+with some non-constant arguments produce a non-constant.  Constants can be used 
+in any context with a source flow which allows a non-constant.  Constants may be 
```

Comment:
I think this sentence covers it, but is it worth elaborating on how `const` composes with `flip`? 
For example, I believe this should work since flow is from `const` to non-`const`:
```firrtl
module FlippedConst:
    input in: {flip x: UInt<1>}
    output out: const {flip x: UInt<1>}
    out <= in
```

Reply:
It should.